tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,286 @@
+// @(#)root/tree:$Id$
+// Author: Luca Giommi   22/08/16
+
+/*************************************************************************
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+#include "TSimpleAnalysis.h"
+#include "TFile.h"
+#include "TCanvas.h"
+#include "TChain.h"
+#include "TH1.h"
+#include "TError.h"
+
+/** \class TSympleAnalysis
+A TSimpleAnalysis object permit to, given an input file or through
+command line, create an .root file in which are saved histograms
+that the user want to create.
+*/
+
+const std::string TSimpleAnalysis::kCutIntr="if";
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case with the input file
+///
+/// \param[in] kFile name of the input file that has to be read
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& kFile)
+{
+   fInputName = kFile;
+   in.open(kFile);
+   if(!in) {
+      ::Error("TSimpleAnalysis","File %s not found",kFile.c_str());
+      throw 1;
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, std::vector<std::string> inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions)
+{
+   fOutputFile=output;
+   fTreeName=name;
+   for (std::string input: inputFiles)
+      fInputFiles.push_back(input);
+   for (std::string expr: expressions) {
```

Comment:
This should be calling `HandlefExpressionConfig` instead of duplicating the code.